ens/main.py

```diff
@@ -58,14 +59,17 @@ def __init__(self, provider=default, addr=None):
         """
         self.web3 = init_web3(provider)
 
-        try:
-            net_version = int(self.web3.net.version)
-            ens_addr = addr if addr else NET_VERSION_TO_ENS_ADDR[net_version]
-        except KeyError:
-            raise UnknownNetwork(
-                "ENS is not available on the current network by default. "
-                "You need to manually set the `addr` parameter to the registry's address."
-            )
+        if addr:
+            ens_addr = addr
+        else:
+            try:
+                net_version = int(self.web3.net.version)
+                ens_addr = NET_VERSION_TO_ENS_ADDR[net_version]
+            except (StaleBlockchain, KeyError):
+                raise UnknownNetwork(
+                    "ENS is not available on the current network by default. "
+                    "You need to manually set the `addr` parameter to the registry's address."
+                )
```

Comment:
I'm not convinced that swallowing `StaleBlockchain` is right here, but won't have a chance to dig in. I would at least recommend `raise from exc` here so that we can see the underlying exception.

Reply:
`StaleBlockchain` is again not handled in my latest commits. It was one of my (failed) attempts to fix the tests.

You can see exception traceback in CI tests of my other commits in this PR.